tensorflow_recommenders_addons/dynamic_embedding/python/ops/tf_save_restore_patch.py

```diff
@@ -527,10 +539,86 @@ def restore(self, sess, save_path):
     except errors.InvalidArgumentError as err:
       # There is a mismatch between the graph and the checkpoint being loaded.
       # We add a more reasonable error message here to help users (b/110263146)
-      raise _wrap_restore_error_with_msg(
+      raise saver._wrap_restore_error_with_msg(
           err, "a mismatch between the current graph and the graph")
 
 
+def _de_keras_save_func(original_save_func,
+                        model,
+                        filepath,
+                        overwrite,
+                        include_optimizer,
+                        signatures=None,
+                        options=None,
+                        save_traces=True,
+                        *args,
+                        **kwargs):
+  """Overwrite TF Keras save function
+    Calling the TF save API for all ranks causes file conflicts, 
+    so KV files other than rank0 need to be saved by calling the underlying API separately.
+    This is a convenience function for saving HvdAllToAllEmbedding to KV files in different rank.
+  """
+  try:
+    import horovod.tensorflow as hvd
+    try:
+      hvd.rank()
+    except:
+      hvd = None
+  except:
+    hvd = None
+
+  call_original_save_func = functools.partial(
+      original_save_func,
+      model=model,
+      filepath=filepath,
+      overwrite=overwrite,
+      include_optimizer=include_optimizer,
+      signatures=signatures,
+      options=options,
+      save_traces=save_traces,
+      *args,
+      **kwargs)
+
+  def _traverse_emb_layers_and_save(hvd_rank):
```

Comment:
Do we have adequate UT cases to cover this function?

Reply:
Done